apps/wallet-dashboard/components/migration/MigrationObjectsPanel.tsx

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+'use client';
+
+import { useFilterMigrationObjects, useGroupedMigrationObjectsByExpirationDate } from '@/hooks';
+import {
+    STARDUST_MIGRATABLE_OBJECTS_FILTER_LIST,
+    STARDUST_UNMIGRATABLE_OBJECTS_FILTER_LIST,
+} from '@/lib/constants';
+import { StardustOutputDetailsFilter } from '@/lib/enums';
+import {
+    Button,
+    ButtonType,
+    Card,
+    CardImage,
+    Chip,
+    ImageShape,
+    InfoBox,
+    InfoBoxStyle,
+    InfoBoxType,
+    Panel,
+    Skeleton,
+    Title,
+} from '@iota/apps-ui-kit';
+import type { IotaObjectData } from '@iota/iota-sdk/client';
+import { Close, Warning } from '@iota/ui-icons';
+import clsx from 'clsx';
+import { useState } from 'react';
+import { MigrationObjectDetailsCard } from './migration-object-details-card';
+import VirtualList from '../VirtualList';
+
+const FILTERS = {
+    migratable: STARDUST_MIGRATABLE_OBJECTS_FILTER_LIST,
+    unmigratable: STARDUST_UNMIGRATABLE_OBJECTS_FILTER_LIST,
+};
+
+interface MigrationObjectsPanelProps {
+    selectedObjects: IotaObjectData[];
+    onClose: () => void;
+    isTimelocked: boolean;
+}
+
+export function MigrationObjectsPanel({
+    selectedObjects,
+    onClose,
+    isTimelocked,
+}: MigrationObjectsPanelProps): React.JSX.Element {
+    const [stardustOutputDetailsFilter, setStardustOutputDetailsFilter] =
+        useState<StardustOutputDetailsFilter>(StardustOutputDetailsFilter.All);
+
+    const {
+        data: resolvedObjects = [],
+        isLoading,
+        error: isErrored,
+    } = useGroupedMigrationObjectsByExpirationDate(selectedObjects, isTimelocked);
+
+    const filteredObjects = useFilterMigrationObjects(resolvedObjects, stardustOutputDetailsFilter);
```

Comment:
We should also sort the filtered objects
With migtatable assets, we have to sort them by expiration time who expire first are the first to show
![image](https://github.com/user-attachments/assets/49f8d7bc-c610-4b0e-907e-fea46039bb7a)

With unmigtatable assets, tokens or native tokens should not appear without an unlock time, right?
![image](https://github.com/user-attachments/assets/763e0de8-2e66-429f-a8d8-075ffac28307)